crates/modelardb_common/src/storage.rs

```diff
@@ -330,6 +335,34 @@ pub async fn extract_azure_blob_storage_arguments(
     Ok((account, access_key, container_name, offset_data))
 }
 
+/// Reinterpret the bits used for univariate ids in `compressed_segments` to convert the column from
+/// [`UInt64Array`] to [`Int64Array`] as the Delta Lake Protocol does not support unsigned integers.
+fn univariate_ids_uint64_to_int64(compressed_segments: &mut Vec<RecordBatch>) {
+    for record_batch in compressed_segments {
+        let mut columns = record_batch.columns().to_vec();
+        let univariate_ids = crate::array!(record_batch, 0, UInt64Array);
+        let signed_univariate_ids: Int64Array =
+            univariate_ids.unary(|value| i64::from_ne_bytes(value.to_ne_bytes()));
+        columns[0] = Arc::new(signed_univariate_ids);
+
+        // unwrap() is safe as columns are constructs to match DISK_COMPRESSED_SCHEMA.
+        *record_batch = RecordBatch::try_new(DISK_COMPRESSED_SCHEMA.0.clone(), columns).unwrap();
+    }
+}
+
+/// Reinterpret the bits used for univariate ids in `compressed_segments` to convert the column from
+/// [`Int64Array`] to [`UInt64Array`] as the Delta Lake Protocol does not support unsigned integers.
+pub fn univariate_ids_int64_to_uint64(compressed_segments: &RecordBatch) -> RecordBatch {
+    let mut columns = compressed_segments.columns().to_vec();
+    let signed_univariate_ids = crate::array!(compressed_segments, 0, Int64Array);
+    let univariate_ids: UInt64Array =
+        signed_univariate_ids.unary(|value| u64::from_ne_bytes(value.to_ne_bytes()));
```

Comment:
Same as above: is it safe to use native endianness? Do we risk to write with one endianness and then later read with another endianess (on the client)?

Reply:
See [this comment](https://github.com/ModelarData/ModelarDB-RS/pull/198#discussion_r1711403886).